### DISCUSSION TITLE
What are the available customization options?

### DISCUSSION THREAD

**Discussion by drorata** [markmap/markmap#123] [Q&A]
I found (based on the web UI) the following options:

- `maxWidth: 300`
- `color: '#2980b9'` and
- `initialExpandLevel: 1`

I could not find where these options are documented nor what are other options?

Specifically, things that I'd like to customize are:
- Spacing
- Shape of nodes (e.g. use a rectangle around each node).

I'm new to `markmap`... Can it be done?

**Comment by gera2ld** [comment]
Currently only the following options are supported in frontmatter:

https://github.com/gera2ld/markmap/blob/16b6efac0b7d82e63567968eddd110ac0afe1377/packages/markmap-common/src/types/common.ts#L101-L108

Spacing can be supported by API, can you explain why you need to change it? Maybe we can add it as an option in the future. 
Shape of nodes are not supported yet.

**Comment by drorata** [comment]
I'm just trying to understand what can be done...

* What's the impact of `duration`? It doesn't seem to have an impact. 
* What can be done using `extraCss`? `extraJs`?
* Regarding the shape of the nodes, I suspect it that enclosing the text in a nice box will make the map more readable.

**Reply by gera2ld** [reply]
- `duration` is the duration of animation in milliseconds, you'll notice the difference by setting `duration: 0` and toggle a node.
- `extraJs`/`extraCss` is used to load additional assets, e.g. third-party plugins, see [demo](https://markmap.js.org/repl#?d=gist:gera2ld/af76a4c245b302206b16aec503dbe07b:katex.md) for [katex mhchem plugin](https://github.com/KaTeX/KaTeX/tree/main/contrib/mhchem)

**Reply by vincent-trupin** [reply]
The demo link working for me is this [one](https://markmap.js.org/repl#?d=gist:af76a4c245b302206b16aec503dbe07b:katex.md)
